tests/unitary/test_withdraw_unitary.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/python3
+
+import brownie
+
+
+# Can user A and user B withdraw in correct proportion?
+def test_withdraw_multiples(multi, base_token, reward_token, alice, bob, charlie, chain):
+    reward_token.approve(multi, 10 ** 16, {"from": alice})
+
+    multi.setRewardsDistributor(reward_token, alice, {"from": alice})
+    multi.notifyRewardAmount(reward_token, 10 ** 15, {"from": alice})
+
+    amount = 10 ** 12
+    base_token.approve(multi, amount, {"from": bob})
+    multi.stake(amount, {"from": bob})
+    bob_stake_timestamp = multi.lastTimeRewardApplicable(reward_token)
+
+    base_token.approve(multi, 10 * amount, {"from": charlie})
+
+    # chain.mine(timedelta=1)
+    multi.stake(amount * 10, {'from': charlie})
+    charlie_stake_timestamp = multi.lastTimeRewardApplicable(reward_token)
+
+    if charlie_stake_timestamp == bob_stake_timestamp:
+        assert multi.earned(charlie, reward_token) // 10 == multi.earned(bob, reward_token)
+    else:
+        assert multi.earned(charlie, reward_token) // 10 <= multi.earned(bob, reward_token)
+    assert multi.balanceOf(charlie) // 10 == multi.balanceOf(bob)
```

Comment:
Couple issues here:

* because there have not been any blocks mined since charlie staked, the value returned from `earned(charlie)` will always be 0
* the `if/else` logic for the comparison based on timestamp feels brittle

Better to check the timestamps, mine a block at some date in the future, and then confirm that the values from `earned` are within an expected distance from one another